app/models/tagset_models/tag_set.rb

```diff
@@ -250,17 +271,21 @@ def remove_from_autocomplete
 
   def add_tags_to_autocomplete(tags_to_add)
     tags_to_add.each do |tag|
+      next if TagSet.invalid_tag?(tag.name) # skips invalid tags in autocomplete
```

Comment:
Due to your changes above, the code never gets to adding tags to the tag set if any tag is invalid - so this extra check is not necessary.